kymatio/scattering1d/filter_bank.py

```diff
@@ -284,33 +292,92 @@ def compute_temporal_support(h_f, criterion_amplitude=1e-3):
         value \\epsilon controlling the numerical
         error. The larger criterion_amplitude, the smaller the temporal
         support and the larger the numerical error. Defaults to 1e-3
+    warn: bool (default False)
+        Whether to raise a warning upon `h_f` leading to boundary effects.
 
     Returns
     -------
     t_max : int
         temporal support which ensures (1) for all rows of h_f
 
     """
+    if h_f.shape[-1] == 1:
+        return 1
+    elif h_f.ndim == 1:
+        h_f = h_f[None]
+
     h = ifft(h_f, axis=1)
     half_support = h.shape[1] // 2
-    # compute ||h - h_[-N, N]||_1
-    l1_residual = np.fliplr(
-        np.cumsum(np.fliplr(np.abs(h)[:, :half_support]), axis=1))
-    # find the first point above criterion_amplitude
-    if np.any(np.max(l1_residual, axis=0) <= criterion_amplitude):
+    # check if any value in half of worst case of abs(h) is below criterion
+    hhalf = np.max(np.abs(h[:, :half_support]), axis=0)
+    max_amplitude = hhalf.max()
+    meets_criterion_idxs = np.where(hhalf <= criterion_amplitude * max_amplitude
+                                    )[0]
+    if len(meets_criterion_idxs) != 0:
         # if it is possible
-        N = np.min(
-            np.where(np.max(l1_residual, axis=0) <= criterion_amplitude)[0])\
-            + 1
+        N = meets_criterion_idxs.min() + 1
+        # in this case pretend it's 1 less so external computations don't
+        # have to double support since this is close enough
+        if N == half_support:
+            N -= 1
     else:
-        # if there is none:
+        # if there are none
         N = half_support
-        # Raise a warning to say that there will be border effects
-        warnings.warn('Signal support is too small to avoid border effects')
+        if warn:
+            # Raise a warning to say that there will be border effects
+            warnings.warn('Signal support is too small to avoid border effects')
+    return N
+
+
+def compute_minimum_required_length(fn, N_init, max_N=None,
```

Comment:
They serve to compute `min_to_pad`, i.e. minimum amount of required padding to avoid boundary effects. 

> just do phi_halfwidth = compute_temporal_support(N)

doesn't achieve this

> just exclude those paths

That's a choice we can make but is a separate consideration. The purpose of this compute block is to tell us the padding we'll need _given_ a configuration.